File: src/main/scala/ex6/Selection.scala
package ex6

import scala.reflect.ClassTag

class Selection(private var a:Array[Int]) extends Sortieren:

    override def sort(): Array[Int] =
        for (i <- 0 until a.length) {
            var min: Int = i
            for (j <- i+1 until a.length) do
                if (apply(j) < apply(min)) then min = j
            swap(i, min)
        }
        countAndReset
        return a

File: src/main/cpp/ex6/Selection.cpp
#include "Selection.hpp"

namespace ex6
{
void Selection::sort()
{
  for (size_t i = 0; i < size - 1; i++)
  {
    size_t min = i;
    for (size_t j = i + 1; j < size; j++)
    {
      if (less(a[j], a[min]))
        min = j;
    }
    swap(i, min);
  }
}
} // namespace ex6
